cli/cmd/logs/logs.go

```diff
@@ -40,33 +40,36 @@ func (l *Logs) Run(ctx *clicontext.CLIContext) error {
 }
 
 func (l *Logs) setupConfig(ctx *clicontext.CLIContext) (*stern.Config, error) {
-	var err error
-	config := &stern.Config{
-		LabelSelector: labels.Everything(),
-		Timestamps:    l.T_Timestamps,
-		Namespace:     ctx.GetSetNamespace(),
+	if len(ctx.CLI.Args()) == 0 {
+		return nil, errors.New("must specify a service or build")
 	}
-
-	config.Template, err = l.logFormat()
+	objName := ctx.CLI.Args().First()
+	obj, err := ctx.ByID(objName)
 	if err != nil {
 		return nil, err
 	}
-
-	config.ContainerState = stern.RUNNING
-	if l.P_Previous {
-		config.ContainerState = stern.TERMINATED
+	if obj.Object == nil {
+		return nil, errors.New("No object found")
 	}
 
-	tail := int64(l.N_Tail)
-	config.TailLines = &tail
+	// handle both build and service lookups
+	var lookupID string
+	if obj.Type == clitypes.BuildType {
+		lookupID = strings.TrimLeft(obj.Name, "taskrun/default/")
```

Comment:
nit: lookupId can be found in object.Name

Reply:
I think this should be changed to use object.Name, default shouldn't be hard-coded.